lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/mic.rb

```diff
@@ -0,0 +1,206 @@
+class Mic
+end# -*- coding: binary -*-
+require 'rex/post/meterpreter'
+require 'bindata'
+
+module Rex
+  module Post
+    module Meterpreter
+      module Ui
+
+        ###
+        #
+        # Mic - Capture audio from the remote system
+        #
+        ###
+        class Console::CommandDispatcher::Stdapi::Mic
+          Klass = Console::CommandDispatcher::Stdapi::Mic
+
+          include Console::CommandDispatcher
+
+          #
+          # List of supported commands.
+          #
+          def commands
+            all = {
+              'mic_start' => 'start capturing an audio stream from the target mic',
+              'mic_stop'  => 'stop capturing audio',
+              'mic_list'  => 'list all microphone interfaces',
+              'listen'    => 'listen to a saved audio recording via audio player'
+            }
+            reqs = {
+              'mic_start' => [ 'audio_mic_start' ],
+              'mic_stop'  => [ 'audio_mic_stop' ],
+              'mic_list'  => [ 'audio_mic_list' ],
+              'listen'    => [ 'audio_mic_start' ]
+            }
+
+            all.delete_if do |cmd, _desc|
```

Comment:
You can replace the rest of this function with `filter_commands(all, reqs)`, e.g https://github.com/rapid7/metasploit-framework/pull/8513/files